src/Host/Protocol/Impl/RUserProfile/RUserProfileServicesImpl.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Security.Principal;
+using System.Text;
+using System.Runtime.InteropServices;
+using Microsoft.Common.Core.OS;
+using Microsoft.Common.Core.Security;
+using Microsoft.Extensions.Logging;
+using static Microsoft.R.Host.Protocol.NativeMethods;
+
+namespace Microsoft.R.Host.Protocol {
+    class RUserProfileServicesImpl : IUserProfileServices {
```

Comment:
I think it would be better to call this class `RUserProfileServices` (since it implements the corresponding interface), and rename the class above. For the latter, so far as I can see, its main purpose is to accept and parse a named pipe request, and delegate to this class for actual execution. So the other class should be named something like `RUserProfileServer`.